### ISSUE TITLE
Invalid syntax in gui/eight_puzzle.py

### ISSUE THREAD

**Issue by LoboLofi** [aimacode/aima-python#896]
I'm trying to run the Gui examples, with:
`python3 -i gui/eight_puzzle.py`
And get the next out:

```
File "gui/eight_puzzle.py", line 88
    b[zero_ix] = Button(root, text=f'{state[index]}', width=6, font=('Helvetica', 40, 'bold'), command=partial(exchange, zero_ix))
                                                   ^
SyntaxError: invalid syntax
```

I've python 3.5.2 and Ubuntu 16.04.

**Comment by apb7** [comment]
Hi @LoboLofi, this isn't a syntax error actually. This feature has been introduced in Python 3.6 and would run smoothly if you update your Python version. Thanks!

**Comment by LoboLofi** [comment]
Thanks a lot @apb7

**Comment by sagar-sehgal** [comment]
@apb7 @ad71 It has been mentioned in the [README](https://github.com/aimacode/aima-python#python-34-and-up) that we can run it with Python3.4 and up.  
So we can still make this compatible for lower version of python by using 
`b[zero_ix] = Button(root, text=state[index], width=6, font=('Helvetica', 40, 'bold'), command=partial(exchange, zero_ix))`
There is a simmilar isssue in [gui/grid_mdp.py](https://github.com/aimacode/aima-python/blob/master/gui/grid_mdp.py) .
This code is running just fine on my system after this change which has simmilar configurations as mentioned above.
Shall I go ahead and create a PR for this? Or is there any gap in my solution?

**Comment by ad71** [comment]
@sagar-sehgal As long as everything still works on Python 3.4+, we're good. You can submit a PR.

**Comment by sagar-sehgal** [comment]
@ad71 @MrDupin  Sorry for replying late but, I submitted th PR for the changes in gui/eight_puzzle.py and gui/genetic_algorithm_example.py  but I didn't do that for gui/grid_mdp.py . Actually  on applying the above changes in grid_mdp.py and then running the code, it was fine until I clicked on a cell. On clicking on a cell and providing it a reward/making it a terminal/making it a cell it was showing a new error
```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.5/tkinter/__init__.py", line 1553, in __call__
    return self.func(*args)
  File "grid_mdp.py", line 128, in update_table
    if wall.get() == WALL_VALUE:
  File "/usr/lib/python3.5/tkinter/__init__.py", line 358, in get
    return self._tk.getint(self._tk.globalgetvar(self._name))
TypeError: getint() argument must be str, not float

```
Further on changing tk.IntVar() to tk.DoubleVar() in line no. 94,96,262 and 264. It was working fine and assigning values on clicking on the cells and changing the color of the cells. 
But on Buiding, it showed another error
```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.5/tkinter/__init__.py", line 1553, in __call__
    return self.func(*args)
  File "grid_mdp.py", line 406, in build
    solve_page.create_graph(gridmdp, terminals, _height, _width)
  File "grid_mdp.py", line 588, in create_graph
    self.canvas.show()
AttributeError: 'FigureCanvasTkAgg' object has no attribute 'show'
```
which was again resolved by changing show() to draw().
And finally it didn't showed up any error. 
Should I submit the PR solving the problem. Or is there anything which I need to do.

Further I would like to show up some of the GUI things which can be improved like 
![screenshot from 2018-12-17 19-22-17](https://user-images.githubusercontent.com/22503629/50091401-4e310800-0231-11e9-80fa-99ea0eb81937.png)
It can be made only 1 selectable at a time (if possible). Further this can also be improved 
![screenshot from 2018-12-17 19-29-17](https://user-images.githubusercontent.com/22503629/50091702-18d8ea00-0232-11e9-9983-11b12e5946b8.png)
we can close the previous window, when a new cell is clicked (if possible).
Should I start working on these things.
